crates/runtime/src/libcalls.rs

```diff
@@ -137,3 +142,195 @@ pub unsafe extern "C" fn wasmtime_imported_memory32_size(
 
     instance.imported_memory_size(memory_index)
 }
+
+/// Implementation of `table.copy` when both tables are locally defined.
+#[no_mangle]
+pub unsafe extern "C" fn wasmtime_table_copy_defined_defined(
```

Comment:
While I realize that you're just persisting the same style as other functions in this module, I think the `#[no_mangle]` here isn't necessary?

Additionally, how come there are 4 table copy functions (permutations of imported/local src/dst) but for memory.copy there's only one? It looks like each of these functions may save one branch when figuring out the table based on the index, but that seems pretty cheap compared to the copy itself.

(seemed a bit wordy to have 4 functions)